akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala

```diff
@@ -524,6 +524,9 @@ private[akka] trait ClusterRouterActor { this: RouterActor =>
     satisfiesRoles(m.roles) &&
     (settings.allowLocalRoutees || m.address != cluster.selfAddress)
 
+  private def memberShuttingDown(event: MemberEvent): Boolean =
+    event.isInstanceOf[MemberPreparingForShutdown] || event.isInstanceOf[MemberReadyForShutdown]
```

Comment:
How does this work for the typed routers? That is via the Receptionist.

Reply:
Yeah, the receptionist provides a separate lists for reachable and then a list with all routees, the group router uses the list of reachable unless that is empty, then falls back to the full list. I guess we could improve that with logic like what exists here.